### ISSUE TITLE
"page not found" after using an internal hyperlink

### ISSUE THREAD

**Issue by Tenavi** [mitmproxy/pdoc#615]
#### Problem Description
If using a hyperlink to something within the package, e.g. `[text to display](submodule1.submodule2.submodule3)`, after clicking on the hyperlink, all links on the linked page return a "page not found" error.

#### Steps to reproduce the behavior:
1. Consider a package with this structure:
```
package
    __init__.py
    submodule1
        __init__.py
       submodule2
            __init__.py
            submodule3
                __init__.py
                ...
```
2. Suppose in the `submodule1` docstring we want to link `package.submodule1.submodule2.submodule3`, but we don't want to write the whole thing so we include a hyperlink:
```
"""
[text to display](submodule1.submodule2.submodule3)
"""
```
3. We run `pdoc`, and click on the hyperlink. `pdoc` correctly links this to the documentation for `submodule3`.
4. All hyperlinks on this page (`package/submodule1/submodule2/submodule3.html`) lose `submodule1/`. For example, the "up one level" button tries to link to `package/submodule2.html`, which doesn't exist so fails.

#### System Information
```
pdoc: 14.0.0
Python: 3.8.5
Platform: macOS-10.16-x86_64-i386-64bit
```

**Comment by mhils** [comment]
I can't reproduce this. Could you please set up a git repo with a clean reproducer?

**Comment by Tenavi** [comment]
> I can't reproduce this. Could you please set up a git repo with a clean reproducer?

Here's a git repo demonstrating the bug. I have `python==3.8.5` in the environment in which the bug was found. Exact instructions are contained in the README and in the documentation generated by using `pdoc` to document the repo.

https://github.com/Tenavi/pdoc_hyperlink

**Comment by mhils** [comment]
Thanks for the excellent reproducer! Manually linking to submodules is unfortunately not supported right now. I've pushed #617 to make sure that the pdoc web server gives you a 404 right away. However, #544 should make it possible to do exactly what you are looking for: You can then use `.submodule2.submodule3` (note the leading dot) in your docstrings and things will be automatically linked. :)

**Comment by mhils** [comment]
Fixed as part of #544 and released in 14.1! :)

**Comment by Tenavi** [comment]
Nice, thanks @mhils !
